### ISSUE TITLE
Indy SDK Getting Started Docker Compose Problem

### ISSUE THREAD

**Issue by tfoutrein** [hyperledger/indy-sdk#2379]
Hello, When trying to deploy the Getting Started Guide with `docker-compose up`

i have this error 
```
 > [5/6] RUN pip3 install -U     pip    ipython-notebook       ipython==7.9     setuptools      jupyter         python3-indy==1.11.0:                                                                                                                                                                                                                                                                                                                                                         
#8 1.445 Collecting pip                                                                                                                                                                                                                                                                                                                                                                                                                                                                       
#8 1.663   Downloading https://files.pythonhosted.org/packages/fe/ef/60d7ba03b5c442309ef42e7d69959f73aacccd0d86008362a681c4698e83/pip-21.0.1-py3-none-any.whl (1.5MB)                                                                                                                                                                                                                                                                                                                         
#8 2.456 Collecting ipython-notebook                                                                                                                                                                                                                                                                                                                                                                                                                                                          
#8 2.698   Could not find a version that satisfies the requirement ipython-notebook (from versions: )                                                                                                                                                                                                                                                                                                                                                                                         
#8 2.713 No matching distribution found for ipython-notebook
#8 2.994 You are using pip version 8.1.1, however version 21.0.1 is available.
#8 2.994 You should consider upgrading via the 'pip install --upgrade pip' command.
------
executor failed running [/bin/sh -c pip3 install -U     pip     ipython-notebook       ipython==7.9     setuptools      jupyter         python3-indy==1.11.0]: exit code: 1
```

Do you have a solution ? 

Thank you in advance

**Comment by xoryouyou** [comment]
I've updated my local Dockerfile to the following
as a workaround.

* upgraded to ubuntu 18.04
* added `libzmq3-dev` `python3-dev` and `build-essential`
* removed `ipython-notebook`


```docker
FROM ubuntu:18.04

RUN useradd -ms /bin/bash indy

# Install environment
RUN apt-get update -y && apt-get install -y \
    wget \
    python3.5 \
    python3-pip \
    python-setuptools \
    apt-transport-https \
    ca-certificates \
    software-properties-common \
    libzmq3-dev \
    python3-dev \
    build-essential

WORKDIR /home/indy

RUN pip3 install -U \
    pip \
    ipython==7.9 \
    setuptools \
    jupyter \
    python3-indy==1.11.0

RUN apt-key adv --keyserver keyserver.ubuntu.com --recv-keys CE7709D068DB5E88 \
    && add-apt-repository "deb https://repo.sovrin.org/sdk/deb xenial stable" \
    && apt-get update \
    && apt-get install -y \
    libindy=1.11.0

USER indy

EXPOSE 8888
```

**Comment by SamYuan1990** [comment]
> I've updated my local Dockerfile to the following
> as a workaround.
> 
> * upgraded to ubuntu 18.04
> * added `libzmq3-dev` `python3-dev` and `build-essential`
> * removed `ipython-notebook`
> 
> ```
> FROM ubuntu:18.04
> 
> RUN useradd -ms /bin/bash indy
> 
> # Install environment
> RUN apt-get update -y && apt-get install -y \
>     wget \
>     python3.5 \
>     python3-pip \
>     python-setuptools \
>     apt-transport-https \
>     ca-certificates \
>     software-properties-common \
>     libzmq3-dev \
>     python3-dev \
>     build-essential
> 
> WORKDIR /home/indy
> 
> RUN pip3 install -U \
>     pip \
>     ipython==7.9 \
>     setuptools \
>     jupyter \
>     python3-indy==1.11.0
> 
> RUN apt-key adv --keyserver keyserver.ubuntu.com --recv-keys CE7709D068DB5E88 \
>     && add-apt-repository "deb https://repo.sovrin.org/sdk/deb xenial stable" \
>     && apt-get update \
>     && apt-get install -y \
>     libindy=1.11.0
> 
> USER indy
> 
> EXPOSE 8888
> ```

I hope we can have any pr to fix and merged.

**Comment by xoryouyou** [comment]
I can try and find some time during the week for a PR if its still needed. :)

**Comment by SamYuan1990** [comment]
> I can try and find some time during the week for a PR if its still needed. :)

I suppose we still need this to be fixed with pr. just meet docker compose problem and fixed with update docker file as workaround above.

**Comment by SamYuan1990** [comment]
```
Building indy_pool
[+] Building 85.8s (15/15) FINISHED                                                                                                                                                                                                                           
 => [internal] load build definition from indy-pool.dockerfile                                                                                                                                                                                           0.0s
 => => transferring dockerfile: 47B                                                                                                                                                                                                                      0.0s
 => [internal] load .dockerignore                                                                                                                                                                                                                        0.0s
 => => transferring context: 2B                                                                                                                                                                                                                          0.0s
 => [internal] load metadata for docker.io/library/ubuntu:16.04                                                                                                                                                                                         85.6s
 => [ 1/11] FROM docker.io/library/ubuntu:16.04@sha256:6a3ac136b6ca623d6a6fa20a7622f098b2fae1ac05f0114386ef439d8ca89a4a                                                                                                                                  0.0s
 => CACHED [ 2/11] RUN apt-get update -y && apt-get install -y  git  wget  python3.5  python3-pip  python-setuptools  python3-nacl  apt-transport-https  ca-certificates  supervisor                                                                     0.0s
 => CACHED [ 3/11] RUN pip3 install -U  pip==9.0.3  setuptools                                                                                                                                                                                           0.0s
 => CACHED [ 4/11] RUN apt-key adv --keyserver keyserver.ubuntu.com --recv-keys CE7709D068DB5E88 ||  apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys CE7709D068DB5E88                                                                  0.0s
 => CACHED [ 5/11] RUN echo "deb https://repo.sovrin.org/deb xenial master" >> /etc/apt/sources.list                                                                                                                                                     0.0s
 => CACHED [ 6/11] RUN useradd -ms /bin/bash -u 1000 indy                                                                                                                                                                                                0.0s
 => CACHED [ 7/11] RUN apt-get update -y && apt-get install -y         python3-pyzmq=18.1.0         indy-plenum=1.12.1~dev989         indy-node=1.12.1~dev1172         python3-indy-crypto=0.4.5         libindy-crypto=0.4.5         python3-orderedse  0.0s
 => CACHED [ 8/11] RUN echo "[supervisord]\nlogfile = /tmp/supervisord.log\nlogfile_maxbytes = 50MB\nlogfile_backups=10\nlogLevel = error\npidfile = /tmp/supervisord.pid\nnodaemon = true\nminfds = 1024\nminprocs = 200\numask = 022\nuser = indy\nid  0.0s
 => CACHED [ 9/11] RUN awk '{if (index($1, "NETWORK_NAME") != 0) {print("NETWORK_NAME = \"sandbox\"")} else print($0)}' /etc/indy/indy_config.py> /tmp/indy_config.py                                                                                    0.0s
 => CACHED [10/11] RUN mv /tmp/indy_config.py /etc/indy/indy_config.py                                                                                                                                                                                   0.0s
 => CACHED [11/11] RUN generate_indy_pool_transactions --nodes 4 --clients 5 --nodeNum 1 2 3 4 --ips="10.0.0.2,10.0.0.2,10.0.0.2,10.0.0.2"                                                                                                               0.0s
 => exporting to image                                                                                                                                                                                                                                   0.0s
 => => exporting layers                                                                                                                                                                                                                                  0.0s
 => => writing image sha256:304d14f32b2e19eb0a88cc0b6b7f32e696050779366dcb2f11c065bec1e6cdf3                                                                                                                                                             0.0s
 => => naming to docker.io/library/indy_pool                                                                                                                                                                                                             0.0s

Use 'docker scan' to run Snyk tests against images to find vulnerabilities and learn how to fix them
Successfully built 304d14f32b2e19eb0a88cc0b6b7f32e696050779366dcb2f11c065bec1e6cdf3
Building jupyter
[+] Building 640.1s (9/11)                                                                                                                                                                                                                                    
 => [internal] load build definition from getting-started.dockerfile                                                                                                                                                                                     0.0s
 => => transferring dockerfile: 803B                                                                                                                                                                                                                     0.0s
 => [internal] load .dockerignore                                                                                                                                                                                                                        0.0s
 => => transferring context: 2B                                                                                                                                                                                                                          0.0s
 => [internal] load metadata for docker.io/library/ubuntu:16.04                                                                                                                                                                                         81.7s
 => [1/8] FROM docker.io/library/ubuntu:16.04@sha256:6a3ac136b6ca623d6a6fa20a7622f098b2fae1ac05f0114386ef439d8ca89a4a                                                                                                                                    0.0s
 => CACHED [2/8] RUN useradd -ms /bin/bash indy                                                                                                                                                                                                          0.0s
 => [3/8] RUN apt-get update -y && apt-get install -y  wget  python3.5  python3-pip  python-setuptools  apt-transport-https  ca-certificates  software-properties-common  libzmq3-dev     python3-dev     build-essential                              553.4s
 => [4/8] WORKDIR /home/indy                                                                                                                                                                                                                             0.0s
 => [5/8] RUN pip3 install --upgrade pip                                                                                                                                                                                                                 4.6s
 => ERROR [6/8] RUN pip3 install -U  pip  ipython-notebook       ipython==7.9  setuptools  jupyter                                                                                                                                                       0.3s 
------                                                                                                                                                                                                                                                        
 > [6/8] RUN pip3 install -U    pip     ipython-notebook       ipython==7.9     setuptools      jupyter:                                                                                                                                                      
#9 0.282 Traceback (most recent call last):                                                                                                                                                                                                                   
#9 0.282   File "/usr/local/bin/pip3", line 7, in <module>                                                                                                                                                                                                    
#9 0.282     from pip._internal.cli.main import main                                                                                                                                                                                                          
#9 0.282   File "/usr/local/lib/python3.5/dist-packages/pip/_internal/cli/main.py", line 57
#9 0.282     sys.stderr.write(f"ERROR: {exc}")
#9 0.282                                    ^
#9 0.282 SyntaxError: invalid syntax
------
executor failed running [/bin/sh -c pip3 install -U 	pip 	ipython-notebook       ipython==7.9 	setuptools 	jupyter]: exit code: 1
ERROR: Service 'jupyter' failed to build
```
error log before it is been resolved. FYI

**Comment by xoryouyou** [comment]
I've made a PR over here https://github.com/hyperledger/indy-sdk/pull/2423
It works on my machine and was only a  bump of the ubuntu version and a change in the pip package name.
Let me know if it works :)

**Comment by apurbapokharel** [comment]
The changes you made works but some of us are getting an error while running the notebook file as mentioned in #2371. Did you encounter the problem as well? If yes then how did you fix it?

**Comment by xoryouyou** [comment]
I haven't read #2371 before and yes my setup also encounters the mentioned loop error.

But with a simple fix I was able to get it running.


![2021-08-31_13-13](https://user-images.githubusercontent.com/777569/131493132-c7ec77b5-56f8-414c-8298-983bb6481485.png)

I don't know if I just got lucky or if this really is a proper fix.

The `await run()` seems to be working for me.

```python
if __name__ == '__main__':
    loop = asyncio.new_event_loop()
    asyncio.set_event_loop(loop)
    await run()
    time.sleep(1)  # FIXME waiting for libindy thread complete
```
